omod/src/main/java/org/openmrs/module/patientqueueing/web/resource/PatientQueueResource.java

```diff
@@ -66,6 +72,38 @@ public PatientQueue getByUniqueId(String uniqueId) {
 		return patientQueue;
 	}
 	
+	@Override
+	public Object update(String uuid, SimpleObject propertiesToUpdate, RequestContext context) throws ResponseException {
+		if (propertiesToUpdate.get("status") == null) {
+			return super.update(uuid, propertiesToUpdate, context);
+		}
+		PatientQueue patientQueue = getPatientQueueForUpdate(uuid, propertiesToUpdate);
+		ValidateUtil.validate(patientQueue);
+		patientQueue.setDateChanged(new Date());
+
+        if (propertiesToUpdate.get("status") != null && propertiesToUpdate.get("status") == "picked") {
```

Comment:
Is there any reason why you have very many `propertiesToUpdate.get("status")` instead of just one?

Reply:
@dkayiwa comment resolved.